packages/surimi/src/types/css-strict.types.ts

```diff
@@ -0,0 +1,164 @@
+import type csstype from 'csstype';
+
+import type { CustomProperty } from '#lib/api/custom-property';
+
+import type { VendorPrefix } from './config.types';
+
+/**
+ * Global CSS keywords that are valid for ANY CSS property.
+ * These are part of the CSS-wide keywords specification.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Values_and_Units#css-wide_values
+ */
+export type GlobalCssValue = 'inherit' | 'initial' | 'unset' | 'revert' | 'revert-layer';
+
+/**
+ * Strict CSS properties with enhanced type hints.
+ * This type extends csstype properties with global CSS keywords for better autocomplete.
+ *
+ * Each property accepts:
+ * - Valid CSS values from csstype (includes all string values, numbers, etc.)
+ * - Global CSS keywords (inherit, initial, unset, revert, revert-layer)
+ * - CustomProperty instances with type hints
+ *
+ * Note: This type provides improved autocomplete and documentation but does not
+ * restrict invalid values at compile time due to TypeScript's structural typing.
+ * It's primarily intended to provide better developer experience through IDE hints.
+ *
+ * @example
+ * ```typescript
+ * const styles: StrictCssProperties = {
+ *   display: 'flex',     // ✅ Gets autocomplete
+ *   color: 'red',        // ✅ Gets autocomplete
+ *   margin: 'inherit',   // ✅ Global value
+ * };
+ * ```
+ */
+export type StrictCssProperties = {
+  [K in keyof csstype.Properties]: csstype.Properties[K] | GlobalCssValue | CustomProperty<csstype.Properties[K]>;
+};
+
+/**
+ * Type for CSS custom properties (--*).
+ * Custom properties can have any name starting with `--` and accept string or number values.
+ *
+ * @example
+ * ```typescript
+ * const customProps: StrictCustomProperties = {
+ *   '--primary-color': '#6366f1',
+ *   '--spacing': '1rem',
+ *   '--opacity': 0.8,
+ * };
+ * ```
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/--*
+ */
+export type StrictCustomProperties = Record<`--${string}`, string | number>;
```

Comment:
Using custom properties like this has the nice advantage of overriding them per block, which surimi properties don't allow right now. But I think we should never mix using surimi properties and these raw CSS properties.
So I think I would rather not have this